### PR TITLE
VOTE-289 Create job to update the NVRF and create a PR

### PR DESCRIPTION
**Pull request by SamiraMSadat**
[VOTE-289 Jira Ticket](https://cm-jira.usa.gov/browse/VOTE-289)

Create a job that pulls the NVRF from the EAC website and replaces the current PDF in the repo.
Check the linked circleci job below.
This will be the first PR to commit the circleci script to stage.

### CODE REVIEW

**Comment by SamiraMSadat** [comment]
@clmedders #54 shows the last PR created by the job. This PR is to merge the script itself into stage.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Use the latest 2.1 version of CircleCI pipeline process engine.
```
Comment (clmedders) [inline .circleci/config.yml:1]: I would remove this line

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+# See: https://circleci.com/docs/configuration-reference
```
Comment (clmedders) [inline .circleci/config.yml:2]: and this line

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+version: 2.1
+orbs:
+  gh: circleci/github-cli@2.2.0
+# Define a job to be invoked later in a workflow.
```
Comment (clmedders) [inline .circleci/config.yml:6]: remove line

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+# See: https://circleci.com/docs/configuration-reference/#jobs
```
Comment (clmedders) [inline .circleci/config.yml:7]: remove line

File: .circleci/config.yml (hunk 2 above)
Comment (clmedders) [inline .circleci/config.yml:2]: I would remove these lines to clean up file... these seem to be out the box suggestions from circle

File: .circleci/config.yml (hunk 4 above)
Comment (clmedders) [inline .circleci/config.yml:7]: Remove these lines

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+jobs:
+  update-nvrf:
+    # Specify the execution environment. You can specify an image from Docker Hub or use one of our convenience images from CircleCI's Developer Hub.
+    # See: https://circleci.com/docs/configuration-reference/#executor-job
```
Comment (clmedders) [inline .circleci/config.yml:11]: Remove lines

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    docker:
+      - image: cimg/base:stable
+    # Add steps to the job
+    # See: https://circleci.com/docs/configuration-reference/#steps
```
Comment (clmedders) [inline .circleci/config.yml:15]: Remove lines

File: .circleci/config.yml [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+      - checkout
+      - gh/install
+      #configure git
+      - run:
+          name: Configure git
+          command: |
+            git config user.email "circleci@df52e93b2563.(none)"
+            git config user.name "Pipeline Commit"
+      # git push origin --delete $NEW_BRANCH_NAME
+      # git checkout -b $NEW_BRANCH_NAME
```
Comment (clmedders) [inline .circleci/config.yml:31]: I beleive you can remove these lines as well since you are doing this later on

File: .circleci/config.yml [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+            -H "Accept: application/json" \
+            -u "$NVRF_PDF_WRITE:")
+            # generate a heredoc in BASH_ENV
+            echo "read -r -d '' STORED_ARTIFACTS \<< 'EOF_ARTIFACTS'" >> $BASH_ENV
+            echo "$artifacts" >> $BASH_ENV
+            echo "EOF_ARTIFACTS" >> $BASH_ENV
+            cd public/files
+            rm $NVRF_FILE_NAME
+            echo $STORED_ARTIFACTS > $NVRF_FILE_NAME
+      # IDEA: if the new pdf is the same as the old PDF, automatically merge PR
```
Comment (clmedders) [inline .circleci/config.yml:55]: I would remove this as well to clean up file

File: .circleci/config.yml [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+      - run:
+          name: Create Pull Request
+          command: |
+            echo $NVRF_PDF_WRITE | tee /tmp/key.txt
+            gh auth login --with-token < /tmp/key.txt
+            gh pr close $CIRCLE_BRANCH
+            gh pr create --title "test pr nvrf" --body "checking to see if this works"
+# Orchestrate jobs using workflows
+# See: https://circleci.com/docs/configuration-reference/#workflows
```
Comment (clmedders) [inline .circleci/config.yml:64]: remove lines

File: .circleci/config.yml [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+      - gh/install
+      #configure git
+      - run:
+          name: Configure git
+          command: |
+            git config user.email "circleci@df52e93b2563.(none)"
+            git config user.name "Pipeline Commit"
+      # git push origin --delete $NEW_BRANCH_NAME
+      # git checkout -b $NEW_BRANCH_NAME
+      #download pdf from
```
Comment (clmedders) [inline .circleci/config.yml:32]: I see what you are doing with these but i think you can remove these type of comments as well. The good thing about circle is the steps help break it down but ok to leave them as well if you feel they are needed.

File: .circleci/config.yml [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+          path: Federal_Voter_Registration_ENG.pdf
+          destination: /pdf-files
+      # Create a variable to store the PDF artifact
+      - run:
+          name: Replace current NVRF
+          command: |
+            artifacts=$(curl -X GET "https://circleci.com/api/v2/project/github/usagov/vote-gov-nvrf-app/$CIRCLE_BUILD_NUM/artifacts" \
+            -H "Accept: application/json" \
+            -u "$NVRF_PDF_WRITE:")
+            # generate a heredoc in BASH_ENV
```
Comment (clmedders) [inline .circleci/config.yml:48]: Is this a needed command? If not I would remove

File: .circleci/config.yml [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+            rm $NVRF_FILE_NAME
+            echo $STORED_ARTIFACTS > $NVRF_FILE_NAME
+      # IDEA: if the new pdf is the same as the old PDF, automatically merge PR
+      - run:
+          name: Create Pull Request
+          command: |
+            echo $NVRF_PDF_WRITE | tee /tmp/key.txt
+            gh auth login --with-token < /tmp/key.txt
+            gh pr close $CIRCLE_BRANCH
+            gh pr create --title "test pr nvrf" --body "checking to see if this works"
```
Comment (clmedders) [inline .circleci/config.yml:62]: I would go on and update tile and body with the text you would want to PR to have

File: .circleci/config.yml (hunk 8 above)
Comment (clmedders) [reply .circleci/config.yml:55]: this is for line 55 for the idea

File: .circleci/config.yml (hunk 10 above)
Comment (clmedders) [reply .circleci/config.yml:32]: this is for lines 30-31

File: .circleci/config.yml (hunk 11 above)
Comment (clmedders) [reply .circleci/config.yml:48]: line 48

File: .circleci/config.yml (hunk 12 above)
Comment (clmedders) [reply .circleci/config.yml:62]: line 62

**Review by clmedders — COMMENTED** [review]
Looks good overall, I see the PR for #54 and it has the PDF in it so looks like the full command is working.  My comments are more for code clean up to make the file more lean.

Just flagging here that we will need to document how to set up and udpated the Github Personal token in casue we need to update that in the future.

**Review by clmedders — CHANGES_REQUESTED** [review]
see comments above

File: .circleci/config.yml (hunk 7 above)
Comment (SamiraMSadat) [reply .circleci/config.yml:31]: Oh yes, this is being done later, good catch

File: .circleci/config.yml (hunk 12 above)
Comment (SamiraMSadat) [reply .circleci/config.yml:62]: Done

**Review by clmedders — APPROVED** [review]

**Pull request merged**